res_partner_operating_unit/models/res_users.py

```diff
@@ -18,16 +18,17 @@ def create(self, vals):
 
     @api.multi
     def write(self, vals):
-        res = super().write(vals)
         if vals.get('default_operating_unit_id'):
             # Add the new OU
             self.partner_id.operating_unit_ids = \
-                [(4, res.default_operating_unit_id.id)]
-        return res
+                [(4, vals.get('default_operating_unit_id'))]
+        return super().write(vals)
```

Comment:
@osi-scampbell I would still write the user first and then update the parner with:
```
    @api.multi
    def write(self, vals):
        res = super().write(vals)
        if vals.get('default_operating_unit_id'):
            # Add the new OU
            self.partner_id.operating_unit_ids = \
                [(4, vals.get('default_operating_unit_id'))]
        return res
```

Reply:
@max3903 If we do that then the check_partner_operating_unit() method runs and throws that error because the we have not written the default ou to the operating_unit_ids first. So we update the partner, then update the user.